crates/esthri/src/ops/download.rs

```diff
@@ -132,7 +132,7 @@ where
         obj_info.parts * obj_info.size as u64
     );
     let dir = init_download_dir(download_path).await?;
-    let mut dest = TempFile::new(dir, None).await?;
+    let mut dest = TempFile::new_with_dir(dir, None).await?;
```

Comment:
so I think the behavior we want is 

- if ESTHRI_TEMP_DIR is unset, we create temp file in the same directory where we are downloading to.
- if ESTHRI_TEMP_DIR is set, write temp files there.

so you might wanna make the field in the config struct optional

```rust
struct Config {
    temp_dir_path: Option<TempDirPath>,
}
```

and then do

```rust
    let dir = init_download_dir(download_path).await?;
    let mut dest = if let Some(dir) = Config::global().temp_dir_path() {
        TempFile::new(dir, None).await?
    } else {
        TempFile::new(dir, None).await?
    };
```